### PR TITLE
fix: sip10 token send form fees bug

### LINKED ISSUE #5415: Sending SIP010 tokens blocked because fee doesn't load
Reproduce by
1. Clicking send in the wallet
2. Select a SIP010 token
3. See error, fee not loading

It doesn't stop showing the loading animation, and the fee never loads. 

https://github.com/leather-wallet/extension/assets/33360391/5cc40797-d356-461a-83e8-6d5a38876d3e

I expect the fee to load... and 
When the fee can't be loaded within 3(?) seconds how about showing a warning about and setting a low fixed fee (0.003 STX) so the user can proceed beyond this point?

### PR DESCRIPTION
**Pull request by alter-eggo**
> Try out Leather build 2ed7544 — [Extension build](https://github.com/leather-wallet/extension/actions/runs/9191178591), [Test report](https://leather-wallet.github.io/playwright-reports/fix/sip10-send-form), [Storybook](https://fix-sip10-send-form--65982789c7e2278518f189e7.chromatic.com), [Chromatic](https://www.chromatic.com/library?appId=65982789c7e2278518f189e7&branch=fix/sip10-send-form)

This pr fixes sip10 send form fees bug, discussed:
https://trustmachines.slack.com/archives/C0520G8J33M/p1716339282877119
https://trustmachines.slack.com/archives/C05LAP952E7/p1716363736626209

## Summary by CodeRabbit

- **New Features**
  - Introduced a new function to retrieve asset contract addresses from contract IDs or fully qualified asset names.

- **Improvements**
  - Updated navigation logic for sending SIP10 tokens to enhance performance and maintainability.
  - Standardized address retrieval in asset string parts for consistency.

- **Refactor**
  - Replaced `Navigate` component with `useNavigate` hook for more efficient navigation handling.

### CODE REVIEW

File: src/app/ui/utils/get-asset-contract-address.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+ *
+ * @param value - the source string: [principal].[contract-name] or [principal].[contract-name]::[asset-name]
+ */
+export const getAssetContractAddress = (value: string): string => {
+  if (value.includes('.')) {
+    return value.split('.')[0];
+  }
+
+  return value;
+};
```
Comment (kyranjamie) [inline src/app/ui/utils/get-asset-contract-address.ts]: Can use fn declaration as we do elsewhere

**Review by kyranjamie — APPROVED** [review]
Not sure this helper belongs in ui, but will refactor out when working on @leather-wallet/stacks

File: src/app/ui/utils/get-asset-contract-address.ts (hunk 1 above)
Comment (kyranjamie) [reply src/app/ui/utils/get-asset-contract-address.ts]: Also should have a unit test

Comment (alter-eggo) [reply src/app/ui/utils/get-asset-contract-address.ts]: fixed

**Comment by 314159265359879** [comment]
With this latest build
image


I can select a token. It doesn't seem to load fees but defaults to fixed low fee for low, medium and high. 
image

When I add an amount that is not available I get no warning about insufficient funds. 

Then I can continue to review 
When clicking confirm and send transaction I see
image

File: src/app/store/transactions/token-transfer.hooks.ts [hunk 2]
```diff
@@ -83,70 +83,79 @@ export function useGenerateFtTokenTransferUnsignedTx(info: Sip10CryptoAssetInfo)
...
-        values && 'memo' in values && values.memo !== ''
-          ? someCV(bufferCVFromString(values.memo || ''))
-          : noneCV();
-
-      const amountAsFractionalUnit = ftUnshiftDecimals(amount, info.decimals || 0);
-      const {
-        address: contractAddress,
-        contractName,
-        assetName,
-      } = getAssetStringParts(info.contractId);
```
Comment (alter-eggo) [inline src/app/store/transactions/token-transfer.hooks.ts:107]: removed this

**Review by kyranjamie — APPROVED** [review]
@314159265359879 to QA

**Comment by markmhendrickson** [comment]
Let's ship? 🚢

**Comment by 314159265359879** [comment]
Retested this version fix/sip10-send-form#40371b5, doesn't work. 

1. When I confirm and send transaction it fails with this error:
image


2. There doesn't seem to be a check for the amount of token
image

**Comment by 314159265359879** [comment]
fee setting by the wallet does look correct now.

**Review by 314159265359879 — APPROVED** [review]
@alter-eggo LGTM!

tested sending with several SIP010 tokens no issues with the latest build: fix/sip10-send-form#2ed7544

**Pull request merged**